tcms/testplans/static/testplans/js/mutable.js

```diff
@@ -0,0 +1,23 @@
+$(document).ready((function () {
+    populateProductVersion();
+
+    $('#id_product').change(populateProductVersion);
+
+    $('#add_id_product').click(function () {
+        return showRelatedObjectPopup(this);
+    });
+
+    $('#add_id_version').click(function () {
+        return showRelatedObjectPopup(this);
+    });
+}));
+
+function populateProductVersion() {
+    let href = $('#add_id_version')[0].href;
+    $('#add_id_version')[0].href = href.slice(0, href.indexOf('&product'));
```

Comment:
this unfortunately doesn't produce the desired effect when the link doesn't contain the '&product' sub-string. As-is the index will be -1 and you end up removing the last character so the link becomes `?_popu&product=x`.

The easiest way to fix this is to add `&product=` to the URL in the template and you will not have to use conditional statements in this function.